### DISCUSSION TITLE
how to use different sphere topologies

### DISCUSSION THREAD

**Discussion by savolla** [pyvista/pyvista#3367] [Q&A]
![Screenshot from 2022-09-26 20-46-07](https://user-images.githubusercontent.com/22548773/192345364-5acc8e49-4e66-49e7-8182-f2e7535a1c9f.png)
in PyVista I only saw the **Radial Toplogy** of a sphere but it's not suitable for my application. I want to use something like **Icosahedron** where all vertices are distributed equally on the sphere. Can I create different topologies like in the image above in PyVista? Thanks!

**Comment by banesullivan** [comment]
This is an awesome feature request. pyvista's `Sphere()` method indeed only generates a radial sphere and is directly from `vtkSphereSource`.  I don't think VTK has any of these other sphere types, but we can easily implement them in Python as I did here: https://stackoverflow.com/a/58575437/10631346

I'll try to follow up later with examples for each of these types

**Reply by savolla** [reply]
thanks!! I also hacked the standard **icosahedron** object and subdivided it by 4 :) and got the result I wanted. 
```python
import pyvista as pv

def create_icosahedron_sphere(center, radius):
    icosahedron = pv.Icosahedron(radius)
    icosahedron_sphere = icosahedron.subdivide(4, subfilter="loop")
    return icosahedron_sphere

mesh = create_icosahedron_sphere((0,0,0), 10)
mesh.plot()
```
![Screenshot from 2022-09-26 21-41-38](https://user-images.githubusercontent.com/22548773/192355430-571f4fae-f791-4c5a-bd8b-5b910a2468e4.png)

This is still not a native PyVista object but a workaround. Subdivision rocks!!
